docker/policyconfiguration/naming.go

```diff
@@ -52,5 +52,20 @@ func DockerReferenceNamespaces(ref reference.Named) []string {
 		}
 		name = name[:lastSlash]
 	}
+
+	portNumColon := strings.Index(name, ":")
+	name = name[:portNumColon]
+	res = append(res, name)
```

Comment:
This should happen only if `name` was changed, otherwise there would be a duplicate value.

… Actually, I’m not at all sure that we should allow non-wildcard $host matching $host:port , historically that did not match. We can safely introduce new semantics for the wildcard syntax, but this might break existing users. So, I’m leaning towards just removing this.

Reply:
I haven't pushed the change yet, but locally I have:

```
        // Needed for TestReferencePolicyConfigurationNamepsaces
        portNumColon := strings.Index(name, ":")
        if portNumColon != -1 {
                name = name[:portNumColon]
                res = append(res, name)
        }
```

`openshift.TestReferencePolicyConfigurationNamepsaces` uses a port number, and it will fail if I include `*.example.com` and `*.com` but will pass on `*.example.com:8443` and `*.com:8443` which isn't something we want IIUC. So, do we keep this, or do we get rid of port number from the test?

Reply:
I think the most compatible behavior is to return `registry.example.com:8443/ns/stream`, `registry.example.com:8443/ns`, `registry.example.com:8443,`*.example.com`, `*.com` — and not `registry.example.com`.

I can see the argument for including `registry.example.com` anyway, even if it changes behavior of existing configurations, for more consistency. It’s a tough call.

Reply:
> I think the most compatible behavior is to return `registry.example.com:8443/ns/stream`, `registry.example.com:8443/ns`, `registry.example.com:8443`,`*.example.com`, `*.com` — and not `registry.example.com`.

Hmm, I guess I will go with this for now. Let me know if you later prefer the one below.

> 
> I can see the argument for including `registry.example.com` anyway, even if it changes behavior of existing configurations, for more consistency. It’s a tough call.

Reply:
so if there is no port number mentioned, for example: `registry.example.com/ns/stream`, should the returned array include `registry.example.com`  in that case? or just `[registry.example.com/ns/stream registry.example.com/ns *.example.com *.com]`

Reply:
Yes; it is included now, so it must be included in the updated version as well.